bucketeer2/src/main/kotlin/io/bucketeer/sdk/android/BKTException.kt

```diff
@@ -55,55 +55,3 @@ sealed class BKTException(
     message
   )
 }
-
-// internal fun Throwable.toBKTException(): BKTException = when (this) {
-//  is StatusRuntimeException -> when (status.code) {
```

Comment:
Json API does not have `Status` equivalent.
I'm not sure how I should use `BKTException`s.

Reply:
For the Rest API, we chose the codes from this list [1]:

```go
// Bucketeer API server
// StatusBadRequest, StatusInternalServerError,
// StatusMethodNotAllowed, StatusNotFound, and StatusUnauthorized

var (
	errContextCanceled   = rest.NewErrStatus(http.StatusBadRequest, "gateway: context canceled")
	errMissingAPIKey     = rest.NewErrStatus(http.StatusUnauthorized, "gateway: missing APIKey")
	errInvalidAPIKey     = rest.NewErrStatus(http.StatusUnauthorized, "gateway: invalid APIKey")
	errInternal          = rest.NewErrStatus(http.StatusInternalServerError, "gateway: internal")
	errInvalidHttpMethod = rest.NewErrStatus(http.StatusMethodNotAllowed, "gateway: invalid http method")
	errTagRequired       = rest.NewErrStatus(http.StatusBadRequest, "gateway: tag is required")
	errUserRequired      = rest.NewErrStatus(http.StatusBadRequest, "gateway: user is required")
	errUserIDRequired    = rest.NewErrStatus(http.StatusBadRequest, "gateway: user id is required")
	errBadRole           = rest.NewErrStatus(http.StatusUnauthorized, "gateway: bad role")
	errDisabledAPIKey    = rest.NewErrStatus(http.StatusUnauthorized, "gateway: disabled APIKey")
	errFeatureNotFound   = rest.NewErrStatus(http.StatusNotFound, "gateway: feature not found")
	errFeatureIDRequired = rest.NewErrStatus(http.StatusBadRequest, "gateway: feature id is required")
	errMissingEventID    = rest.NewErrStatus(http.StatusBadRequest, "gateway: missing event id")
	errMissingEvents     = rest.NewErrStatus(http.StatusBadRequest, "gateway: missing events")
	errBodyRequired      = rest.NewErrStatus(http.StatusBadRequest, "gateway: body is required")
)
```

[1] https://go.dev/src/net/http/status.go

Reply:
Thanks, will implement a converter in the next ApiClient PR.

Reply:
Thank you!